types/atom/index.d.ts

```diff
@@ -4482,6 +4485,9 @@ export interface Project {
     /** Invoke a callback when a filesystem change occurs within any open project path. */
     onDidChangeFiles(callback: (events: FilesystemChangeEvent) => void): Disposable;
 
+    /** Invoke a callback whenever the project's configuration has been replaced. */
+    onDidReplace(callback: (projectSpec: ProjectSpecification | undefined) => void): Disposable;
```

Comment:
Sorry, one minor tweak. Looking [at the spec for project specification](https://github.com/atom/atom/pull/16845/files#diff-9e8e46b4759f3f15410bf17731812f46R280), `projectSpec` also can be `null`